new-docs/content/any/getting-started/policies/index.md

```diff
@@ -9,325 +9,438 @@ description: Learn about writing Oso policies - the source of truth for authoriz
+so policies containing tens or hundreds of thousands of rules like
+that can still be used efficiently should your application and
+policy require it, but it's not really recommended.
+
+What we usually want instead is to write rules that match more
+than one argument by exploiting regularities or abstractions in
+our application objects and policy. For instance, each of the three
+rules above has the same second and third parameters, so we can
+collapse all three rules into one if we _conditionally_ match the
+first argument. We can do that by _binding_ a _variable_ called
+`actor` and checking whether its value matches any of the three
+allowed names:
+
+```polar
+allow(actor, "read", "resource") if
+    actor = "Abagail" or
+    actor = "Carol" or
+    actor = "Johann";
+```
 
-Take the following example. We want a rule saying that accountants can read
-expenses. Our initial version might look like:
+We would read this in English as: "allow an actor to read resource
+if the actor is Abagail, or it is Carol, or is Johann". Admittedly,
+this doesn't seem to buy us much over what we had, but we can refactor
+again using the built-in list membership operator `in`:
 
 ```polar
-allow(user: User, "read", expense: Expense) if
-    user.role = "accountant";
+allow(actor, "read", "resource") if
+    actor in ["Abagail", "Carol", "Johann"];
 ```
 
-This would be fine, but if, for example, we wanted to allow the CFO to do
-whatever an accountant can do, we would need to duplicate all the rules. Or if
-we want to change how an application determines roles we would need to change
-all locations using this.
+Now we have a single rule that can be easily modified if our
+allowed list of users changes.
 
-So instead, we can refactor the role check into its own rule:
+## Instances and Fields
 
-```polar
-allow(user: User, "read", expense: Expense) if
-    role(user, "accountant");
+We still have an explicit enumeration of permissions in the rule
+above. That's fine for users represented by strings and such, but
+most applications use more structured representations for their
+actors, actions, and resources.
 
-role(user, role_name) if
-    user.role = role_name;
-```
+Suppose then that our actors are represented by instances of a
+{{< lang >}} `User` class, with, say, a user ID and administrator
+flag:
 
-`role(user, "accountant")` is another example of pattern matching in Polar. Any
-time a rule body contains a **predicate** like this, it is performing another
-query. That is, it will try to find all *matching* rules named `role` with two
-inputs.
+```python
+@dataclass
+class User:
+    id: int
+    admin: bool
+```
 
-You can also either use the [REPL](repl) or the `Oso.query_rule()` method to
-interact with this directly. For example:
+Let's also assume also a simple `Resource` class, with an `owner`
+field that references a user ID:
 
 ```python
-from oso import Oso
+@dataclass
+class Resource:
+    owner: int
+```
```

Comment:
Consistency check: if we change "resource" above to be "document:ID", we can change this class to be a `Document` class as well.